src/games/rota/game.ts

```diff
@@ -0,0 +1,166 @@
+import { IGameArgs, IGameCtx, INVALID_MOVE } from 'boardgame.io/core';
+
+// players are either allowed to place or move pieces
+export enum Phase {
+  Place = 'Place',
+  Move = 'Move',
+}
+
+export interface Point {
+  x: number;
+  y: number;
+  playerID: string;
+  pieceID: number;
+}
+
+export interface IG {
+  points: Point[];
+  piecesPlaced: number;
+  selectedPoint: Point;
+}
+
+function findPoint(G: IG, pointX: number, pointY: number): Point {
+  for (const p of G.points) {
+    if (p.x === pointX && p.y === pointY) {
+      return p;
+    }
+  }
+  return { x: null, y: null, playerID: null, pieceID:null };
+}
+
+function isVictory(G: IG): boolean {
+  for (const p of G.points) {
+    if (p.playerID !== null) {
+      const axisVar: number[] = [-1, 0, 1];
+      // check along x-axis
+      let testWin = true;
+      for (const y of axisVar) {
+        testWin = testWin && p.playerID === findPoint(G, p.x, y).playerID;
+      }
+      if (testWin) {
+        return true;
+      }
+
+      // check along x-axis
+      testWin = true;
+      for (const x of axisVar) {
+        testWin = testWin && p.playerID === findPoint(G, x, p.y).playerID;
+      }
+      if (testWin) {
+        return true;
+      }
+
+      // check Corners Points
+      if (p.x !== 0 && p.y !== 0) {
+        // test around the circumference
+        testWin = p.playerID === findPoint(G, 0, p.y).playerID;
+        testWin = testWin && p.playerID === findPoint(G, p.x, 0).playerID;
+        if (testWin) {
+          return true;
+        }
+
+        // test diagonal
+        testWin = p.playerID === findPoint(G, 0, 0).playerID;
+        testWin = testWin && p.playerID === findPoint(G, -p.x, -p.y).playerID;
+        if (testWin) {
+          return true;
+        }
+      }
+    }
+  }
+  return false;
+}
+
+function placePiece(G: IG, ctx: IGameCtx, pointID: number) {
+  // if more than 6 pieces are placed, do nothing
+  let piecesPlaced = G.piecesPlaced;
+  if (piecesPlaced >= 6) {
+    return { ...G };
+  }
+
+  // add player-id of player who clicked the point
+  const points = G.points.map((point, idx) => {
+    if (point.playerID === null && pointID === idx) {
+      piecesPlaced = piecesPlaced + 1;
+      return { ...point, playerID: ctx.currentPlayer, pieceID: G.piecesPlaced };
+    } else {
+      return { ...point };
+    }
+  });
+
+  return { ...G, points, piecesPlaced };
+}
+
+export function movePiece(G: IG, ctx: IGameCtx, currentID: number, newID: number): IG | string {
+  if (
+    G.points[currentID].playerID === null ||
```

Comment:
isnt this redundant with ` G.points[currentID].playerID !== ctx.playerID` given that ctx.playerID isnt null?